### PR TITLE
Feature storm if regex

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: synapse/lib/ast.py [hunk 1]
```diff
@@ -2478,6 +2480,10 @@ async def expr_or(x, y):
     return await tobool(x) or await tobool(y)
 async def expr_and(x, y):
     return await tobool(x) and await tobool(y)
+async def expr_re(x, y):
+    if regex.search(await tostr(y), await tostr(x), flags=regex.IGNORECASE):
```
Comment (vEpiphyte) [inline synapse/lib/ast.py:2484]: Would we also want to do multiline here as well?

Comment (invisig0th) [reply synapse/lib/ast.py:2484]: Probably whatever makes it most "grep like" so, we may wanna enable multi-line evaluation.  We'll probably eventually have legit regex instances and such.  May not want IGNORECASE as the default.

File: synapse/lib/ast.py [hunk 2]
```diff
@@ -2486,6 +2492,7 @@ async def expr_and(x, y):
     '/': expr_div,
     '=': expr_eq,
     '!=': expr_ne,
+    '~=': expr_re,
```
Comment (invisig0th) [inline synapse/lib/ast.py:2495]: Could we scope this to add `^=` as well while we're here?

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/ast.py [hunk 3]
```diff
@@ -2478,6 +2480,15 @@ async def expr_or(x, y):
     return await tobool(x) or await tobool(y)
 async def expr_and(x, y):
     return await tobool(x) and await tobool(y)
+async def expr_prefix(x, y):
+    x, y = await tostr(x), await tostr(y)
+    # y = await tostr(y)
```
Comment (invisig0th) [inline synapse/lib/ast.py:2485]: remove plz :)

**Review by invisig0th — CHANGES_REQUESTED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**